kyo-core/shared/src/main/scala/kyo/ios.scala

```diff
@@ -1,25 +1,28 @@
 package kyo
 
+import IOs.internal.*
 import core.*
-import core.internal.*
-import iosInternal.*
 import java.util.concurrent.atomic.AtomicReference
 import scala.annotation.tailrec
 import scala.util.*
 import scala.util.control.NonFatal
 
-sealed trait IOs extends Effect[IOs]:
+opaque type IOs <: Defers = Defers & SideEffects
 
-    private val tag = Tag[IOs]
-
-    val unit: Unit < IOs = ()
+object IOs:
 
     inline def apply[T, S](
         inline f: => T < (IOs & S)
     ): T < (IOs & S) =
-        new KyoIO[T, S]:
-            def apply(v: Unit, s: Safepoint[IOs & S], l: Locals.State) =
-                f
+        SideEffects.suspend[Unit, T, Defers & S]((), _ => f: T < (Defers & SideEffects & S))
+
+    inline def unit: Unit < IOs = ()
+
+    def run[T](v: T < IOs)(using f: Flat[T < IOs]): T =
+        runLazy(v).pure
+
+    def runLazy[T: Flat, S](v: T < (IOs & S)): T < S =
+        Defers.run(SideEffects.handle(handler)((), v))
 
     def fail[T](ex: Throwable): T < IOs =
```

Comment:
pre-existing - shouldn't this return `Nothing`?

Reply:
good catch! I've fixed it